### ISSUE TITLE
[BUG] Failed public DNS query from the cluster - curl Process exited with status 5

### ISSUE THREAD

**Issue by maxmoser1** [crc-org/crc#3800]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it? **Yes**
  * Running CRC on: Laptop

## CRC version
```bash
CRC version: 2.24.1+81aeeb
OpenShift version: 4.13.6
Podman version: 4.4.4
```
  
## CRC status
```bash
crc status --log-level debug
time="2023-08-05T18:58:06+02:00" level=warning msg="Failed to add proxy CA to crc http transport"
DEBU CRC version: 2.24.1+81aeeb
DEBU OpenShift version: 4.13.6
DEBU Podman version: 4.4.4
DEBU Running 'crc status'
CRC VM:          Running
OpenShift:       Unreachable (v4.13.6)
RAM Usage:       6.61GB of 15.34GB
Disk Usage:      16.39GB of 32.68GB (Inside the CRC VM)
Cache Usage:     22.7GB
Cache Directory: C:\Users\max\.crc\cache
```

## CRC config
```bash
- consent-telemetry                     : no
- cpus                                  : 8
- http-proxy                            : http://user@domain:PASS...@1.2.3.4:8080
- https-proxy                           : http://user@domain:PASS...@1.2.3.4:8080
- ingress-http-port                     : 8080
- memory                                : 15000
- nameserver                            : 10.0.0.1
- proxy-ca-file                         : c:\temp\proxy.crt
```

## Host Operating System
```bash
Microsoft Windows 10 Enterprise
10.0.19044 Build 19044
```

### Steps to reproduce

```
crc start  --log-level debug
```

### Expected

`crc start` completes without warnings about public DNS.

### Actual

```
INFO Check internal and public DNS query...
DEBU Running SSH command: curl --head quay.io --proxy http://user@domain:PASS...@1.2.3.4:8080 --noproxy 127.0.0.1,localhost --insecure --proxy-insecure
DEBU SSH command results: err: Process exited with status 5, output:
WARN Failed public DNS query from the cluster: ssh command error:
command : curl --head quay.io --proxy http://user@domain:PASS...@1.2.3.4:8080 --noproxy 127.0.0.1,localhost --insecure --proxy-insecure
err     : Process exited with status 5
 :
```

Maybe this has todo with the proxy user format?  "user@domain:Pass.." insteat of "user:Pass.."?

### Logs

https://gist.github.com/maxmoser1/903373202dda41558a6b9a9eb6be11cc

**Comment by praveenkumar** [comment]
@maxmoser1 we pass the proxy setting as it is without escaping any special char, it is user responsibility to escape any special char if required. https://www.cyberciti.biz/faq/unix-linux-export-variable-http_proxy-with-special-characters/ contains detail around using hex code.

**Comment by anjannath** [comment]
```
Failed to update cluster proxy configuration: Failed to add proxy cert details ssh command error:
command : timeout 30s oc apply -f /tmp/user-ca-bundle.json --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1
: error: error parsing /tmp/user-ca-bundle.json: json: offset 1425: invalid character '\r' in string literal
```
from the debug logs, it is failing at the end because of #3785 which was fixed in #3799
